graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rest/PipelineResource.java

```diff
@@ -323,6 +317,20 @@ public PipelineSource routing(@ApiParam(name = "body", required = true) @NotNull
         return pipelineSource;
     }
 
+    @ApiOperation(value = "Remove a stream routing rule from the default routing pipeline.")
+    @Path("/remove_routing")
+    @PUT
+    @AuditEvent(type = PipelineProcessorAuditEventTypes.PIPELINE_UPDATE)
+    public Response deleteRouting(@ApiParam(name = "body", required = true) @NotNull DeleteRoutingRequest request) throws NotFoundException {
+        checkPermission(RestPermissions.STREAMS_EDIT, request.streamId());
+        checkPermission(PipelineRestPermissions.PIPELINE_RULE_CREATE);
+
+        String ruleName = routingRuleName(request.inputId(), request.streamId());
```

Comment:
Is there perhaps a better way to do this? We could provide the rule ID to the frontend so that the /system/pipelines/rule DELETE endpoint can be used. We should also delete the rule from the pipeline. I have noticed that this is not handled properly for /system/pipelines/rule DELETE either.